### ISSUE TITLE
Yolo 26 Knowladge Distillation

### ISSUE THREAD

**Issue by Fthgzlkr** [ultralytics/ultralytics#24678]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I wanted to apply Knowledge Distillation (KD) to the YOLOv26m model using a YOLOv26x model as the teacher. Is there any documentation or code Implemention available for yolo 26 . I found implements for older yolo models can I use them on yolo 26 I found the following repos : 
- https://github.com/KefanZhan/YOLOv8-KD 
- https://github.com/danielsyahputra/yolo-distiller

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Fthgzlkr, thank you for your interest in Ultralytics 🚀! This is an automated response to help get your issue routed quickly, and an Ultralytics engineer will also assist you soon.
We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python) and [CLI](https://docs.ultralytics.com/usage/cli) usage examples and where many of the most common questions may already be answered.
If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.
If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips).
For your knowledge distillation question specifically, please share a bit more detail about your setup, such as:
- Your current `ultralytics` version
- Whether you are modifying the training loop or using an external implementation
- The exact model pair you want to use for teacher/student
- Any errors, logs, or code snippets from your current attempt
Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.
## Upgrade
Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:
```bash
pip install -U ultralytics
```
## Environments
YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>
## Status
<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>
If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes) and [Tasks](https://docs.ultralytics.com/tasks) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by glenn-jocher** [comment]
There isn’t an official YOLO26 knowledge distillation implementation or dedicated training doc in the Ultralytics repo yet, and older YOLOv8 KD repos are not drop-in compatible with YOLO26 because the training/loss pipeline differs, so if you want to try it today the practical route is a custom PyTorch teacher-student loop around Ultralytics models; for background, see the [knowledge distillation overview](https://www.ultralytics.com/glossary/knowledge-distillation).

**Comment by lmycross** [comment]
We will release knowledge distillation support for YOLO26 soon, please stay tuned.

**Comment by UltralyticsAssistant** [comment]
A potential fix for this issue has now been merged in [PR #24894: Add knowledge distillation for YOLO detection models](https://github.com/ultralytics/ultralytics/pull/24894).

This update may resolve the issue through several related changes, including:
- added built-in knowledge distillation training support via `distill_model` and `dis`
- introduced a `DistillationModel` wrapper to manage teacher/student training, feature hooks, projector layers, and combined loss handling
- added resume support for distillation checkpoints, including automatic teacher model reconstruction
- updated checkpoint saving so only the student model is exported
- improved trainer behavior for distillation, including safer handling of compile, frozen teacher parameters, and extra `dis_loss` logging
- added tests and documentation for distillation across detect, segment, pose, and obb workflows

Credit to @lmycross for the PR, with contributions from @onuralpszr, @glenn-jocher, and @Laughing-q.

If you'd like to test whether this resolves the issue, you can install the latest `main` branch with:

```bash
pip install git+https://github.com/ultralytics/ultralytics.git@main
```

Or wait for the next official PyPI release if you prefer a packaged release build.

Please let us know whether the merged changes resolve the issue in your setup. Thanks 🙏 for reporting this, and any further feedback is welcome if the issue still persists.
